### ISSUE TITLE
eps

### ISSUE THREAD

**Issue by lqfjun88** [ETSSmartRes/VAE-NILM#1]
Dear Sir,
Thank you for sharing the codes. I m a Chinese researcher and am also researching on power disaggregation.
I tried running the codes and found some problems.
I think the problem is because of missing eps inputs when generating the dataset
Could u advice what should be the values? I think is 16 float vector
regards
qingfan

**Comment by ETSSmartRes** [comment]
Hello Qingfan,

The "eps" variable needs to be set as input in the model because it is declared as an input tensor :
eps = tf.keras.Input(tensor=K.random_normal(shape=(K.shape(x)[0], latent_dim)), name="eps")

However, it corresponds to a random tensor (generator), then you don't need to put a value for it when you fit the model.

Hope that will help you.

Antoine

**Comment by lqfjun88** [comment]
Understand, thanks for the reply!
regards
Qingfan




发件人：ETSSmartRes ***@***.***>
发送日期：2021-10-21 22:21:37
收件人：ETSSmartRes/VAE-NILM ***@***.***>
抄送人：lqfjun88 ***@***.***>,Author ***@***.***>
主题：Re: [ETSSmartRes/VAE-NILM] eps (Issue #1)

Hello Qingfan,
The "eps" variable needs to be set as input in the model because it is declared as an input tensor :
 eps = tf.keras.Input(tensor=K.random_normal(shape=(K.shape(x)[0], latent_dim)), name="eps")
However, it corresponds to a random tensor (generator), then you don't need to put a value for it when you fit the model.
Hope that will help you.
Antoine
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ETSSmartRes** [comment]
Hello Qingfan,

I understood the problem, it appears with Tensorflow 2.

To solve the problem you can replace the line:

`eps = tf.keras.Input(tensor=K.random_normal(shape=(K.shape(x)[0], latent_dim)), name="eps")`

with:
```
def get_eps(x, latent_dim):
    eps = K.random_normal(shape=(K.shape(x)[0], latent_dim))
    return eps
                    
eps = tf.keras.layers.Lambda(get_eps, arguments={"latent_dim": latent_dim})(x)
```

and change:

`model = tf.keras.Model(inputs=[x, eps], outputs=[x_pred])`

for

`model = tf.keras.Model(inputs=[x], outputs=[x_pred])`

It should solve the problem on Tensorflow 2+

Antoine

**Comment by lqfjun88** [comment]
Thanks Antoine. Yes, I m using tf v2 and was unable to execute.

**Comment by RengarWang** [comment]
> 

Thank you. I’m using tensorflow2+.Your method has also solved my problem. 

But it occurred another error: "TypeError: An op outside of the function building code is being passed a "Graph" tensor".

I tried to solve the problem,Disable eager by using the following functions_ excution：
`tf.compat.v1.disable_eager_execution()`

I also saw this problem reflected on stackoverflow:
<https://stackoverflow.com/questions/59403281/unable-to-use-mse-of-vgg-features-in-loss-function>

I hope it works for you.
